integration-tests/velocity/src/test/java/org/apache/camel/quarkus/component/velocity/it/VelocityTest.java

```diff
@@ -0,0 +1,205 @@
+        assertEquals(OLD_BODY, result.get("in.body"));
+        assertTrue(result.containsKey("result_value"));
+        assertEquals(BODY, result.get("result_value"));
+        assertTrue(result.containsKey("body"));
+        assertEquals(NEW_BODY, result.get("body"));
+    }
+
+    @Test
+    public void testBodyAsDomainObject() {
+        RestAssured.given()
+                .queryParam("name", "Sheldon")
+                .queryParam("country", "Earth 1")
+                .contentType(ContentType.JSON)
+                .body(new Person("Sheldon", "Earth 2"))
+                .post("/velocity/bodyAsDomainObject") //
+                .then()
+                .statusCode(201)
+                .body(equalTo("\nHi Sheldon from Earth 2"));
+    }
+
+    @Test
+    public void testValuesInProperties() {
+        RestAssured.given() //
+                .queryParam("name", "Sheldon")
+                .queryParam("item", "1")
+                .contentType(ContentType.TEXT)
+                .body("Dear ${exchange.properties.name}. You ordered item ${exchange.properties.item}.")
+                .post("/velocity/withProperties") //
+                .then()
+                .statusCode(201)
+                .body(equalTo("Dear Sheldon. You ordered item 1."));
+    }
+
+    @Test
+    public void testContentCacheFalse() throws Exception {
+        testContentCache(false);
+    }
+
+    @Test
+    public void testContentCacheTrue() throws Exception {
+        testContentCache(true);
+    }
+
+    private void testContentCache(boolean useContentCache) throws Exception {
+        File template = createFile("velocity_test", "Hi ${body}");
+
+        RestAssured.given()
+                .queryParam("template", "file:/" + template.getPath())
+                //                .queryParam("loaderCache", "false")
+                .queryParam("contentCache", useContentCache)
+                .contentType(ContentType.TEXT)
+                .body("Sheldon")
+                .post("/velocity/template") //
+                .then()
+                .statusCode(201)
+                .body(equalTo("Hi Sheldon"));
+
+        //override file
+        Files.write(Paths.get(template.getPath()), "Bye ${body}".getBytes());
+
+        RestAssured.given() //
+                .queryParam("template", "file:/" + template.getPath())
+                //                .queryParam("loaderCache", "false")
+                .queryParam("contentCache", useContentCache)
+                .contentType(ContentType.TEXT)
+                .body("Sheldon")
+                .post("/velocity/template") //
+                .then()
+
+                .statusCode(201)
+                .body(equalTo(useContentCache ? "Hi Sheldon" : "Bye Sheldon"));
+    }
+
+    private File createFile(String fileName, String body) throws IOException {
+        File tmpFile = File.createTempFile(fileName, ".vm");
+        FileWriter writer = new FileWriter(tmpFile);
+        writer.write(body);
+        writer.close();
```

Comment:
If you use a writer, it needs to be enclosed in a try-with. You also implicitly use the current machines default encoding, which can be just anything - the test might fail on machines with exotic default encodings. Shorter & solving all named problems: 
```suggestion
        Files.write(tmpFile.toPath(), body.getBytes(StandardCharsets.UTF_8);
```

Reply:
I missed that one (copy paste), I'll fix it

Reply:
done